.drone.yml

```diff
@@ -13,25 +13,67 @@ steps:
     settings:
       mode: lint
       chart: ./helm
-      values_file: ./helm/values.yaml
+      values_files:
+        - ./helm/values/dev-values.yaml
 
   - name: deploy to dev
     image: pelotech/drone-helm3
     settings:
       namespace: callisto-dev
       mode: upgrade
       chart: ./helm
-      values_file: ./helm/values.yaml
+      values_files:
+        - ./helm/values/dev-values.yaml
       release: callisto-ingress-nginx
-      kube_certificate: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUMyRENDQWNDZ0F3SUJBZ0lSQUxBdk1yVGVzQkhNUkliR0ZZa3JsaHN3RFFZSktvWklodmNOQVFFTEJRQXcKRlRFVE1CRUdBMVVFQXhNS2EzVmlaWEp1WlhSbGN6QWVGdzB4TnpBNE1qZ3hPVFU0TlRSYUZ3MHlOekE0TWpneApPVFU0TlRSYU1CVXhFekFSQmdOVkJBTVRDbXQxWW1WeWJtVjBaWE13Z2dFaU1BMEdDU3FHU0liM0RRRUJBUVVBCkE0SUJEd0F3Z2dFS0FvSUJBUURQNm5ZVGpmbmR2T2kxOEltYm5idk9aOXJ3c1hxSkZFTE1IVTkxT2VIaXFIT2kKT1luVFZ3cmpCaDJpZExUM0ozbkVoYnM0WFRIcVlwcVEzdmcwVzluVVRPMnk5ZmFmdlRZVTk0NGZXUTl1SXJOZwo1Y0Q0SXhzK05vWTJmdkkrcStYUUM2SFNiRGJ0S0dOUy9Kbmc5SFdsazF2dzY0TlBwbU5IMTJtUFVvNGpGZnc5CmZoL3RwY1k5akVsZnhJckJBVEJzNHhYcDJHOFFrK1ZCSWE5bnF4bmtVTHJvT0t2Y092T3g4RDU1TXdXcEVKc3QKdENZSXFHM05xdlhMcUdTQzdueHoxYmJNSmZjRDdBRmdvZnM1bXZ0bmlRTHBnZG93WmxNc3g2cnJVM2pBUkw3OApvTktXR3p3c3BKNjhybXdhSGZ4NlRBWldHbUxSU2NPd0xlbWJ3cDRIQWdNQkFBR2pJekFoTUE0R0ExVWREd0VCCi93UUVBd0lCQmpBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUEwR0NTcUdTSWIzRFFFQkN3VUFBNElCQVFDb2RrZkMKS2NOYzY5cXR6ZllWUUFlajFVb0hhMjhUUzlqMHV3RVdqWlU0dFVWR2cvZDNCK2YvcnYyN2tqRGM3U2JJSForVQpxMlpMVlVnYU85OVoyUjQyUFhNVTZCRUVUdjUvT0t4amdkNlFNQnFFVkxBdU1XZFpsbnBkMEpDMDFGd0tJb3BGCnVpVUFoWnRjR0tJRXg1U1N1SUlKdFU4a2pnR2xuS0V6aXJrUVVCOUJUOURSQ1VhelQ1ZUp6VFVLSVliWjBNL3gKSUZMbUQvZmZZeFhmRGlSM25YWkdZSWpkUGJHYW42ZU5pZExjQkdEZkxabzZFdkJTR2FUSHl5KzEyc3c2clhTVwpBTVFWd3BUY2ZpSlM1NG9xSXhDU3dyam5XNFZYRnhiSC96SWZXMGs5M29ucERNdXhjT20rN0NuZjViRlJEdlpzCjRCaEV1ank1eEVtajA1TWwKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
+      kube_certificate: 
+        from_secret: kube_certificate
       kube_api_server:
-        from_secret: dev_kube_api_url
+        from_secret: not_prod_kube_api_url
       kube_token:
         from_secret: dev_kube_token
     when:
       event: push
       branch: main
 
+
+---
+kind: pipeline
+type: kubernetes
+name: callisto-ingress-test-deploy
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+  - name: lint helm files
+    image: pelotech/drone-helm3
+    settings:
+      mode: lint
+      chart: ./helm
+      values_files:
+        - ./helm/values/test-values.yaml
+
+  - name: deploy to test
+    image: pelotech/drone-helm3
+    settings:
+      namespace: callisto-test
+      mode: upgrade
+      chart: ./helm
+      values_files:
+        - ./helm/values/test-values.yaml
+      release: callisto-ingress-nginx
+      kube_certificate: 
+        from_secret: kube_certificate
+      kube_api_server:
+        from_secret: not_prod_kube_api_url
+      kube_token:
+        from_secret: test_kube_token
+    when:
+      event: push
```

Comment:
Are we not promoting deployments?

Reply:
I thought we were only promoting for the actual services where we're packaging builds.. 
Given these are more like 'one time'  env specific value changes for ingress I thought we'd want it to deploy on main.
Perhaps i misunderstood, so should we be in fact promoting these deployments?

Reply:
Hey Stephen, it hadn’t actually occurred to us to have manual promotion gates in our ingress config, but I can definitely see the value in having a mechanism for ensuring ingress changes look ok in our not-prod environments before applying them to prod

Daleel and I reckon it would be overkill to have promotion steps between _all_ of our environments. We propose that for all environments other than prod, the changes are applied immediately after merge. We don’t want to be in a situation where we have to do, say, 4 different manual promotions for different environments every time we make a change (including rolling back).

Reply:
I was thinking less about the manual promotion that we have built into the service and more about the release process.
Ideally the promote to test should depend on the deploy to dev pipeline. If something was to be broken in a change, currently we would break both env. I would advocate we deploy to dev and when the pipeline successfully completes it will deploy to test. If deploying to dev fails, it should not deploy to test. I'd probably pee myself in excitement if the deployment could actually validate itself as well i.e. curl the ingress point to make sure we didn't break anything but I may have got ahead of myself

Reply:
> I'd probably pee myself in excitement
😂 wonderful

That makes a lot of sense 👍  do you agree that a manual promote step before going to prod is a sensible idea while our automation isn’t yet something to get dangerously excited about?

Reply:
Agreed, that makes sense. @stephenkaye-ho Regarding curling the ingress point for verification could we potentially add a new ticket to cover that which could maybe be pickup during some random downtime along the journey of this ticket and just depend on the dev pipeline success for now?

Reply:
I'm happy that we add verification as a new ticket. Regarding whether or not we have a manual promote step, I'd look back to why we added one to start with. I hope I get this right. The manual promote step was there to allow for QA to validate the work before it was deployed i.e. run automated or manual tests. Would they do this with the ingress configuration? Maybe one we would feel more comfortable ourselves having a manual step to allow us to verify the deployment before going to prod.
For me deploying sequentially through the environments increased my confidence and is easy to implement.

Reply:
Following conversation with @daleelR @holly-blues. We agreed that the following would be new tickets and therefore out of the scope of this PR.
1. Verifying deployments i.e. curl the endpoint
2. Manual promotion to production.

This PR will be updated to make the deployments for dev and test sequential